pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -82,7 +82,8 @@ type ServiceMonitorSpec struct {
 	Endpoints []Endpoint `json:"endpoints"`
 
 	// Label selector to select the Kubernetes `Endpoints` objects to scrape metrics from.
-	Selector metav1.LabelSelector `json:"selector"`
+	Selector EndpointSelector `json:"selector"`
```

Comment:
Reading @bboreham's [comment](https://github.com/prometheus-operator/prometheus-operator/issues/5345#issuecomment-2451684926), I assume that the request is to offer a way to generate a config which is

````
  kubernetes_sd_configs:
  - role: endpoints
    selectors:
    - role: service
      label: foo=bar
    - role: endpoints
      label: foo=bar
  relabel_configs:
  - ...
````

instead of the current

```
  kubernetes_sd_configs:
  - role: endpoints
  relabel_configs:
  - ...
  - action: keep
    source_labels:
    - __meta_kubernetes_service_label_foo
    - __meta_kubernetes_service_labelpresent_foo
    regex: (bar);true
  - ...
```

Reply:
friendly reminder @simonpasquier

Reply:
@simonpasquier PTAL 🙏🏽

Reply:
@simonpasquier @slashpai friendly ping